lib/codebreaker/classes/game.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# codebreaker
+module Codebreaker
+  include Errors
+  # game entity
+  class Game
+    DIFFICULTIES = { easy: { attempts: 15, hints: 2 },
+                     medium: { attempts: 10, hints: 1 },
+                     hell: { attempts: 5, hints: 1 } }.freeze
+
+    attr_reader :player, :difficulty, :attempts, :hints, :secret_code
+
+    include Uploader
+    extend Validator
+
+    def initialize(player, difficulty)
+      @player = player
+      @difficulty = difficulty.to_sym
+      @attempts = DIFFICULTIES[@difficulty][:attempts]
+      @hints = DIFFICULTIES[@difficulty][:hints]
+      @secret_code = generate_secret_code
+      @list_of_hints = @secret_code.digits.reverse
+    end
+
+    def self.valid?(difficulty)
+      validate_difficulty?(difficulty)
+    end
+
+    def self.guess_valid?(guess)
+      validate_guess?(guess)
+    end
+
+    def take_hint
+      raise Codebreaker::HintsError if @hints.zero?
+
+      random_index = rand(0...@list_of_hints.size)
+      hint = @list_of_hints[random_index]
+      @list_of_hints.delete_at(random_index)
+      @hints -= 1
+      hint
+    end
+
+    def check_the_guess(guess)
+      @attempts -= 1
+      @code_guess = zipped(guess.to_i.digits)
+      check = { exect_hit: exect_hit, wrong_position_hit: wrong_position_hit, empty_hit: 0 }
+      check[:empty_hit] += empty_hit(check)
+      check
+    end
+
+    def create_stats
```

Comment:
Please use external service for statistic and uploading

Reply:
Done